pkg/resources/management.cattle.io/v3/clusterproxyconfig/validator.go

```diff
@@ -0,0 +1,87 @@
+package clusterproxyconfig
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/rancher/webhook/pkg/admission"
+	webhookadmission "github.com/rancher/webhook/pkg/admission"
+	controllerv3 "github.com/rancher/webhook/pkg/generated/controllers/management.cattle.io/v3"
+	admissionv1 "k8s.io/api/admission/v1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/utils/trace"
+)
+
+var gvr = schema.GroupVersionResource{
+	Group:    "management.cattle.io",
+	Version:  "v3",
+	Resource: "clusterproxyconfigs",
+}
+
+// Validator for validating clusterproxyconfigs.
+type Validator struct {
+	admitter admitter
+}
+
+// NewValidator returns a new validator for clusterproxyconfigs.
+func NewValidator(cpsCache controllerv3.ClusterProxyConfigCache) *Validator {
+	return &Validator{
+		admitter: admitter{
+			cpsCache: cpsCache,
+		},
+	}
+}
+
+// GVR returns the GroupVersionKind for this CRD.
+func (v *Validator) GVR() schema.GroupVersionResource {
+	return gvr
+}
+
+// Operations returns list of operations handled by this validator.
+func (v *Validator) Operations() []admissionregistrationv1.OperationType {
+	return []admissionregistrationv1.OperationType{admissionregistrationv1.Create}
+}
+
+// ValidatingWebhook returns the ValidatingWebhook used for this CRD.
+func (v *Validator) ValidatingWebhook(clientConfig admissionregistrationv1.WebhookClientConfig) []admissionregistrationv1.ValidatingWebhook {
+	valWebhook := admission.NewDefaultValidatingWebhook(v, clientConfig, admissionregistrationv1.NamespacedScope, v.Operations())
+	valWebhook.FailurePolicy = admission.Ptr(admissionregistrationv1.Ignore)
```

Comment:
Is there any specific reason why we want to set `FailurePolicy` to `Ignore`? This can lead to multiple `ClusterProxyConfig` for the same downstream cluster if the webhook is temporarily unavailable.  I would leave the default value (`Fail`).

Reply:
Thanks. I agree.